packages/ts-codegen/src/__tests__/index.test.ts

```diff
@@ -744,6 +744,310 @@ describe(generateTS, () => {
     });
   });
 
+  it("import an entity, which is used to implement interface and returned by resolver", () => {
+    let { resolvers, models } = runGenerateTest(graphql`
+      extend schema @import(from: "@msteams/packages-test", defs: ["Entity"])
+      
+      interface Person implements Entity {
+          id: ID!
+      }
+
+      type User implements Person {
+        id: ID!
+      }
+
+      extend type Query {
+        userById(id: ID!): Person
+      }
+    `);
+    expect(models).toMatchInlineSnapshot(`
+      "import type { EntityModel } from "@msteams/packages-test";
+      // Base type for all models. Enables automatic resolution of abstract GraphQL types (interfaces, unions)
+      export interface BaseModel {
+          __typename: string;
+      }
```

Comment:
Nit: Why are we writing with 4 spaces instead of 2? I think we use 2 everywhere, so would be nice to do that here too.